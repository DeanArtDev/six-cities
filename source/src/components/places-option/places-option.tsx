import * as React from 'react';
import {OffersSortTypes} from '@root/types';

const sortTypes = Object.values(OffersSortTypes);

type Props = {
  onTypeSort: (sortType: string) => void;
  typeSort: string;
  isShow: boolean;
}

export const PlacesOption: React.FC<Props> = (props: Props) => {
  const {onTypeSort, typeSort = sortTypes[0], isShow = false} = props;

  return (
    <ul
      className={`places__options places__options--custom${isShow ? ` places__options--opened` : ``}`}
      onClick={(evt: React.MouseEvent) => {
        const target = evt.target as HTMLElement;
        if (target.tagName === `LI`) {
          onTypeSort(target.dataset.sort);
        }
      }}
    >
      {sortTypes.map((type) => {
        return (
          <li
            className={`places__option${typeSort === type ? ` places__option--active` : ``}`}
            data-sort={type}
            key={(+new Date() * Math.random())}
            tabIndex={0}
          >
            {type}
          </li>
        );
      })}
    </ul>
  );
};
