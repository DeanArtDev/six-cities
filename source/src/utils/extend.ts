const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
