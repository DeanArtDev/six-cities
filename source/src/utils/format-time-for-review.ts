import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
