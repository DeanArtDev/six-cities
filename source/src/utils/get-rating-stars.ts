const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
