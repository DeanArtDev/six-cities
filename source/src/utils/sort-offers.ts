import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
