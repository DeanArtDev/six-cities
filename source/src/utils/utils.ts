export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
