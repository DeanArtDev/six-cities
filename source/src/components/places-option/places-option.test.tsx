import {PlacesOption} from './places-option';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {noop} from '@root/utils/utils';

describe(`<PlacesOption/>: `, () => {

  it(`should render opened`, () => {
    const tree = renderer
      .create(
          <PlacesOption
            typeSort={``}
            onTypeSort={noop}
            isShow={true}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render closed`, () => {
    const tree = renderer
      .create(
          <PlacesOption
            typeSort={``}
            onTypeSort={noop}
            isShow={false}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
