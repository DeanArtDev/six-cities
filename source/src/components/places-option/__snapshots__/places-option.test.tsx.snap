// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlacesOption/>:  should render closed 1`] = `
<ul
  className="places__options places__options--custom"
  onClick={[Function]}
>
  <li
    className="places__option"
    data-sort="Popular"
    tabIndex={0}
  >
    Popular
  </li>
  <li
    className="places__option"
    data-sort="Price: low to high"
    tabIndex={0}
  >
    Price: low to high
  </li>
  <li
    className="places__option"
    data-sort="Price: high to low"
    tabIndex={0}
  >
    Price: high to low
  </li>
  <li
    className="places__option"
    data-sort="Top rated first"
    tabIndex={0}
  >
    Top rated first
  </li>
</ul>
`;

exports[`<PlacesOption/>:  should render opened 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
  onClick={[Function]}
>
  <li
    className="places__option"
    data-sort="Popular"
    tabIndex={0}
  >
    Popular
  </li>
  <li
    className="places__option"
    data-sort="Price: low to high"
    tabIndex={0}
  >
    Price: low to high
  </li>
  <li
    className="places__option"
    data-sort="Price: high to low"
    tabIndex={0}
  >
    Price: high to low
  </li>
  <li
    className="places__option"
    data-sort="Top rated first"
    tabIndex={0}
  >
    Top rated first
  </li>
</ul>
`;
